File: March 2023/Day 17/Number of Subarrays of 0's/number-of-subarrays-of-0s.cpp
//{ Driver Code Starts
// Initial Template for C++

#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
// User function Template for C++
// Time Complexity :- O(n);
// Space Compleixty :- O(1);
long long int no_of_subarrays(int n, vector<int> &arr) {
    // Write your code here.
    long long int ct=0,ans=0;
	for(int v : arr) {
	    if(v==0) {
	        ct++;
	    } else {
	        ans+=(ct*(ct+1)/2);
	        ct=0;
	    }
	}
	ans+=(ct*(ct+1)/2);
	return ans;
}

//{ Driver Code Starts.

int main() {

    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        vector<int> arr(n);
        for (int i = 0; i < n; i++) {
            cin >> arr[i];
        }
        cout << no_of_subarrays(n, arr) << endl;
    }
}

// } Driver Code Ends

File: March 2023/Day 17/Number of Subarrays of 0's/number-of-subarrays-of-0s.py
#User function Template for python3

class Solution():
    # Time Complexity :- O(n);
    # Space Compleixty :- O(1);
    def no_of_subarrays(self, n,arr):
        #your code goes here
        ct=0;
        ans=0;
		for v in arr :
		    if v==0 :
		        ct+=1;
		    else :
		        ans+=(ct*(ct+1)//2);
		        ct=0;
		ans+=(ct*(ct+1)//2);
		return ans;


#{ 
 # Driver Code Starts
#Initial Template for Python 3


for _ in range(int(input())):
    n = int(input())
    arr = [int(i) for i in input().split()]
    print(Solution().no_of_subarrays(n, arr))
# } Driver Code Ends